abstract_specification.md

```diff
@@ -122,6 +122,68 @@ SOMACollection is an unordered, `string`-keyed map of values. Values may be any
 - as a container of independent objects (e.g., a collection of single-cell datasets, each manifest as a [SOMAExperiment](#soma-experiment) object)
 - as the basis for building other composed types (e.g., using SOMACollection to organize pre-defined fields in [SOMAExperiment](#soma-experiment) such as multiple layers of `X`).
 
+#### Collection entry URIs
+
+Collections refer to their components by URI.
+A collection may store its reference to an element by **absolute** URI or **relative** URI.
+
+A reference to an **absolute** URI is, as the name implies, absolute.
+When the backing storage of the collection itself is moved, absolute URI references do not change.
+
+A collection can also refer to sub-paths of itself by **relative** URI.
+If the entire directory storing a collection and a child referred to by relative URI is moved, the relative URI references now refer to the changed base URI.
```

Comment:
you might add some wording here that points out that "relative" is an attribute of the particular storage system, eg. S3 really doesn't have directories, but has path prefixes that serve the same purpose.  TileDB-Cloud shows another bunch of special cases if I understand it right.

So I would point out that relative is optional.

And I would reword this to something like:    Where relative URIs are supported by the storage system, collections referencing children by a relative URI will always resolve the child with respect to the parent (base) URI.  For example, on a Posix file system, if the entire directory storing a collection .... <same as you have>

Reply:
Added some description of this, along with language indicating that we treat “fake” hierarchies just like normal hierarchies.